mne/preprocessing/eog.py

```diff
@@ -249,3 +250,112 @@ def create_eog_epochs(raw, ch_name=None, event_id=998, picks=None, tmin=-0.5,
                         reject_by_annotation=reject_by_annotation,
                         decim=decim)
     return eog_epochs
+
+
+def eog_regression(inst, eog_evokeds=None, eog_channels=None, picks=None):
+    """Remove EOG signals from the EEG channels by regression.
+
+    Employs linear regression to remove EOG signals from other channels, as
+    described in [1]_. Optionally, one may chose to use evoked blink/saccade
+    data to obtain the regression weights as described in [2]_.
+
+    The operation is performed in-place.
+
+    Parameters
+    ----------
+    inst : Raw | Epochs
+        The data on which the EOG correction produce should be performed.
+    eog_evokeds : Evoked | list of Evoked | None
+        Optional recordings of averaged eye movements. For example averaged
+        blinks and/or saccades (use a list to supply different kinds of
+        movements). When specified, regression weights will be fitted on this
+        data, before being applied to the data given as ``inst`` parameter.
+        Regression weights obtained from averaged data might be more robust
+        [2]_.
+    eog_channels : str | list of str | None
+        The names of the EOG channels to use in the regression. By default, all
+        EOG channels are used.
+    picks : list of int | None
+        Channels from which to remove the EOG. By default, the correction is
+        applied to EEG channels only.
+
+    Returns
+    -------
+    inst : Raw | Epochs
+        The version of the data with the EOG removed.
+    weights : ndarray, shape (n_eog_channels, n_channels)
+        The regression weights. For each EOG channel, the fitted regression
+        weight to each data channel. The ordering of the weights matches the
+        ordering of the channels of the object given as ``inst`` parameter.
+    intercept : ndarray, shape (n_channels,)
+        For each data channel, the fitted intercept. The ordering of the
+        intercepts matches the ordering of the channels of the object given as
+        ``inst`` parameter.
+
+    References
+    ----------
+    .. [1] Gratton, G. and Coles, M. G. H. and Donchin, E. (1983). A new method
+           for off-line removal of ocular artifact. Electroencephalography and
+           Clinical Neurophysiology, 468-484.
+           https://doi.org/10.1016/0013-4694(83)90135-9
+    .. [2] Croft, R. J. and Barry, R. J. (1998). EOG correction: a new
+           aligned-artifact average solution. Clinical Neurophysiology, 107(6),
+           395-401. http://doi.org/10.1016/s0013-4694(98)00087-x
+    """
+    # Handle defaults for EOG channels parameter
+    if eog_channels is None:
+        eog_picks = pick_types(inst.info, meg=False, ref_meg=False, eog=True)
+        eog_channels = [inst.ch_names[ch] for ch in eog_picks]
```

Comment:
Whatever logic is here should be deduplicated with what's done in `compute_ssp_eog` (or whatever it's called) and put in a private function

Reply:
Turns out that was already done. TIL about the `_get_eog_channel_index` function.